frame/support/procedural/src/lib.rs

```diff
@@ -445,6 +445,28 @@ pub fn construct_runtime(input: TokenStream) -> TokenStream {
 /// pallet. Otherwise it implements `StorageInfoTrait` for the pallet using the
 /// `PartialStorageInfoTrait` implementation of storages.
 ///
+/// ## Dev Mode (`#[pallet(dev_mode)]`)
+///
+/// Specifying the argument `dev_mode` will allow you to enable dev mode for a pallet. The aim
+/// of dev mode is to loosen some of the restrictions and requirements placed on production
+/// pallets for easy tinkering and development. Dev mode pallets should not be used in
+/// production. Enabling dev mode has the following effects:
+///
+/// * Weights no longer need to be specified on every `#[pallet::call]` declaration. By default, dev
+///   mode pallets will assume a weight of zero (`0`) if a weight is not specified. This is
+///   equivalent to specifying `#[weight(0)]` on all calls that do not specify a weight.
+/// * All storages are marked as unbounded, meaning you do not need to implement `MaxEncodedLen` on
+///   storage types. This is equivalent to specifying `#[pallet::unbounded]` on all storage type
+///   definitions.
+///
+/// <div class="example-wrap" style="display:inline-block"><pre class="compile_fail"
```

Comment:
Right now AFAIK there is no built-in way of doing a warning block unless something about the syntax of your code creates a doc warning (which isn't the case here), so the common workaround is to just use the same CSS classes rust docs use when there is a real warning and hard-code the HTML for now.

https://github.com/rust-lang/rust/issues/73935 has tracked it since 2020

The actual HTML I use here was based on an SO answer that I am now having trouble locating, but yes it is using some of the built-in classes to achieve this warning look